### PR TITLE
Display x-to-many relation preview values in content manager list

### PR DESCRIPTION
**Pull request by HichamELBSI**
### What does it do?

Display a tooltip to preview the value of the x-to-many relations.

### Why is it needed?

Describe the issue you are solving.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

Capture d’écran 2020-12-02 à 17 56 42

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import Tooltip from '../Tooltip';
+
+const RelationPreviewTooltip = ({ tooltipId, rowId, mainField, name }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [relationData, setRelationData] = useState([]);
+  const { slug } = useListView();
+  const tooltipRef = useRef();
+
+  const fetchRelationData = useCallback(async () => {
+    const requestURL = getRequestUrl(`collection-types/${slug}/${rowId}/${name}`);
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js:18]: Can you try to add the maximum needed information into the layout? In the editview the layout regarding the relations includes the following:

```
layouts: {
  editRelations: [
    {
      fieldSchema: {},
      metadatas: {},
      name: 'categories',
      queryInfos: {
        endPoint: '/content-manager/relations/application::address.address/categories',
        containsKey: 'published_at_contains',
        defaultParams: {},
      },
    },
  ],
},
```

I think that using the same format for the relation, might help us later with i18n as we will have a more granular control of the endpoint, especially if we want to change the endpoint.

Here I would like us to generate the `queryInfos` info object for the relations (so you need to update the `formatLayout` and maybe the `formatLayoutToApi` utils) so you can add the static part of the endpoint `collection-types/${slug}/`. By doing so, you will also remove the dependency to the `useListView` hook.

File: packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      setIsLoading(false);
+    } catch (err) {
+      console.error({ err });
+      setIsLoading(false);
+    }
+  }, [mainField, name, rowId, slug]);
+
+  useEffect(() => {
+    // temp : Should remove the setTimeout and fetch the data
+    const timeout = setTimeout(() => {
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js:36]: Can you just add the cancel subscriptions code even though we are not fetching data yet?

```
const abortController = new AbortController();
// ...

return () => {
  clearTimeout(timeout);
  abortController.abort();
}
```

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js:18]: I don't really understand why we need the `containsKey: published_at_containes` here?

File: packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import getRequestUrl from '../../utils/getRequestUrl';
+import getMockData from './mockData';
+import Tooltip from '../Tooltip';
+
+const RelationPreviewTooltip = ({ tooltipId, rowId, mainField, name }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [relationData, setRelationData] = useState([]);
+  const tooltipRef = useRef();
+  const abortController = new AbortController();
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/components/RelationPreviewList/RelationPreviewTooltip.js:14]: Can you declare this constant inside your effect and pass the signal to the `fetchReflationData` so you don't have to pass the signal as a dependency of your effect?

**Review by soupette — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**